library/leakcanary-sample/src/main/java/com/example/leakcanary/ExampleApplication.java

```diff
@@ -32,5 +33,19 @@ public static RefWatcher getRefWatcher(Context context) {
   @Override public void onCreate() {
     super.onCreate();
     refWatcher = LeakCanary.install(this);
+    enabledStrictMode();
+  }
+
+  private void enabledStrictMode() {
+    StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder() //
+        .detectAll() //
+        .penaltyLog() //
+        .penaltyDeath() //
+        .build());
+    StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder() //
+        .detectAll() //
```

Comment:
Excellent point. I actually looked at Strict mode when I started working on LeakCanary, I liked the idea but hated the implementation.